embassy-executor/src/arch/wasm.rs

```diff
@@ -73,9 +73,10 @@ mod thread {
         pub fn start(&'static mut self, init: impl FnOnce(Spawner)) {
             unsafe {
                 let executor = &self.inner;
-                self.ctx.closure.write(Closure::new(move |_| {
+                let future = Closure::new(move |_| {
                     executor.poll();
-                }));
+                });
+                self.ctx.closure.write_in_place(|| future);
```

Comment:
This code is technically fine to just use `write` since the [`Closure` struct is only small](https://docs.rs/wasm-bindgen/0.2.87/src/wasm_bindgen/closure.rs.html#241-244), but I figure it's safest to remove `write` and instead make all callers go through `write_in_place`, since futures are _in general_ of unbounded size, and leaving a more convenient `write` function in place could cause this bug to slip back in as the code changes in future.